internal/scheduler/nodedb_test.go

```diff
@@ -0,0 +1,739 @@
+package scheduler
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	"github.com/G-Research/armada/internal/scheduler/schedulerobjects"
+)
+
+func createNodeDb(nodes []*schedulerobjects.Node) (*NodeDb, error) {
+	db, err := NewNodeDb(testPriorities, testResources)
+	if err != nil {
+		return nil, err
+	}
+	err = db.Upsert(nodes)
+	if err != nil {
+		return nil, err
+	}
+	return db, nil
+}
+
+// testNodeItems1 has max of 1Gb and 7cpu available, so check that such jobs requesting less than this
+// can be scheduled
+func TestSelectNodeForPod_SimpleSuccess(t *testing.T) {
+	for i := 1; i < 7; i++ {
+		testName := fmt.Sprintf("cpu %d", i)
+		t.Run(testName, func(t *testing.T) {
+			db, err := createNodeDb(testNodeItems1)
+			assert.NoError(t, err)
+			report, err := db.SelectAndBindNodeToPod(uuid.New(), &schedulerobjects.PodRequirements{
+				Priority: 0,
+				ResourceRequirements: v1.ResourceRequirements{
+					Requests: v1.ResourceList{
+						"cpu":    resource.MustParse(strconv.Itoa(i)),
+						"memory": resource.MustParse("1Gi"),
+					},
+				},
+			})
+			assert.NoError(t, err)
+			assert.NotNil(t, report.Node)
+		})
+	}
+}
+
+// testNodeItems1 has max of 1Gb and 7cpu available, so check that such jobs requesting more than this
+// cant be scheduled
+func TestSelectNodeForPod_SimpleCantSchedule(t *testing.T) {
+	db, err := createNodeDb(testNodeItems1)
+	assert.NoError(t, err)
+
+	invalidResources := []v1.ResourceList{
+		{"cpu": resource.MustParse("8"), "memory": resource.MustParse("1Gi")},
+		{"cpu": resource.MustParse("1"), "memory": resource.MustParse("10Gi")},
+		{"cpu": resource.MustParse("8000Mi")},
+	}
+
+	for _, r := range invalidResources {
+		report, err := db.SelectAndBindNodeToPod(uuid.New(), &schedulerobjects.PodRequirements{
+			Priority:             0,
+			ResourceRequirements: v1.ResourceRequirements{Requests: r},
+		})
+		assert.NoError(t, err)
+		assert.Nil(t, report.Node)
+	}
+}
+
+// Test that some resource we don't know about causes an error:
+// TODO:  Is returning an error here correct?
```

Comment:
we get an  error in this test- is that correct?

Reply:
Should be fixed now